File: alu/vk/src/main/kotlin/AluTypes.kt
import verik.base.*
import verik.data.*

val DATA_WIDTH = 8

@alias fun t_UbitData() = t_Ubit(DATA_WIDTH)

enum class Op {
    ADD,
    SUB,
    AND,
    OR,
    XOR,
    SLT,
    SLTU,
    SLL,
    SRL,
    SRA
}

File: alu/ch/src/main/scala/alu/AluTypes.scala
package alu

import chisel3.util._

object Const {

  val DATA_WIDTH = 8
}

object Op {

  val add :: sub :: and :: or :: xor :: slt :: sltu :: sll :: srl :: sra :: Nil = Enum(10)
  val WIDTH = add.getWidth
}
